notebooks/qwen2.5-omni-chatbot/gradio_helper.py

```diff
@@ -0,0 +1,207 @@
+            else:
+                lines[i] = "<br></code></pre>"
+        else:
+            if i > 0:
+                if count % 2 == 1:
+                    line = line.replace("`", r"\`")
+                    line = line.replace("<", "&lt;")
+                    line = line.replace(">", "&gt;")
+                    line = line.replace(" ", "&nbsp;")
+                    line = line.replace("*", "&ast;")
+                    line = line.replace("_", "&lowbar;")
+                    line = line.replace("-", "&#45;")
+                    line = line.replace(".", "&#46;")
+                    line = line.replace("!", "&#33;")
+                    line = line.replace("(", "&#40;")
+                    line = line.replace(")", "&#41;")
+                    line = line.replace("$", "&#36;")
+                lines[i] = "<br>" + line
+    text = "".join(lines)
+    return text
+
+
+def _remove_image_special(text):
+    text = text.replace("<ref>", "").replace("</ref>", "")
+    return re.sub(r"<box>.*?(</box>|$)", "", text)
+
+
+def is_video_file(filename):
+    video_extensions = [".mp4", ".avi", ".mkv", ".mov", ".wmv", ".flv", ".webm", ".mpeg"]
+    return any(filename.lower().endswith(ext) for ext in video_extensions)
+
+
+def is_audio_file(filename):
+    audio_extensions = [".mp3", ".wav", "flac", ".m4a", ".wma"]
+    return any(filename.lower().endswith(ext) for ext in audio_extensions)
+
+
+def transform_messages(original_messages):
+    transformed_messages = []
+    for message in original_messages:
+        new_content = []
+        for item in message["content"]:
+            if "image" in item:
+                new_item = {"type": "image", "image": item["image"]}
+            elif "text" in item:
+                new_item = {"type": "text", "text": item["text"]}
+            elif "video" in item:
+                new_item = {"type": "video", "video": item["video"]}
+            elif "audio" in item:
+                new_item = {"type": "audio", "audio": item["audio"]}
+            else:
+                continue
+            new_content.append(new_item)
+
+        new_message = {"role": message["role"], "content": new_content}
+        transformed_messages.append(new_message)
+
+    return transformed_messages
+
+
+def make_demo(model, processor):
+    def call_local_model(model, processor, messages):
+        messages = transform_messages(messages)
+
+        text = processor.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)
+        audios, images, videos = process_mm_info(messages, use_audio_in_video=False)
+        inputs = processor(text=text, audio=audios, images=images, videos=videos, padding=True, return_tensors="pt")
+
+        tokenizer = processor.tokenizer
+        gen_kwargs = {"thinker_max_new_tokens": 1024, "return_audio": False, **inputs}
+        streamer = TextIteratorStreamer(tokenizer, timeout=60.0, skip_prompt=True, skip_special_tokens=True)
```

Comment:
in some slow platforms 60.0 timeout may be too high for first inference with big input (e.g. video processing)

```suggestion
        streamer = TextIteratorStreamer(tokenizer, skip_prompt=True, skip_special_tokens=True)
```